node.go

```diff
@@ -179,15 +185,38 @@ func (t *nodeIterator) Done() bool {
 	case element.Bucket != nil:
 		t.bucket = element.Bucket
 		t.bucketIndex = 1
-
 		t.next = &(*element.Bucket)[0]
 		return false
+	case element.HashMapNode != nil:
+		childNode, err := t.ls.Load(
+			ipld.LinkContext{Ctx: context.TODO()},
+			*element.HashMapNode,
+			HashMapNodePrototype,
+		)
+		if err != nil {
+			t.doneErr = fmt.Errorf("failed to load child node from linksystem: %w", err)
+			return false
+		}
+		child, ok := bindnode.Unwrap(childNode).(*HashMapNode)
+		if !ok {
+			t.doneErr = fmt.Errorf("failed to unwrap child node: %w", err)
+			return false
+		}
+		t.parents = append(t.parents, nodeIteratorStep{node: child})
+		return t.Done()
 	default:
-		panic(fmt.Sprintf("unexpected element type: %v", element))
+		t.doneErr = fmt.Errorf("unexpected element type: %v", element)
```

Comment:
There's also the case of `element.Bucket != nil && element.HashMapNode != nil`, which maybe should be checked first and a similar error to this generated, it shouldn't be allowed to have both. Kind of like this: https://github.com/filecoin-project/go-hamt-ipld/blob/d7766f8628ec68928eca53ffc9a9e27f80b67d4e/hamt.go#L386-L390 (in fact the single case could be stated the same way: `(element.Bucket == nil) == (element.HashMapNode == nil)` - that's legal Go isn't it?)